Documentation/pinning.md

```diff
@@ -0,0 +1,74 @@
+# Object pinning
+
+BPF has a persistent view of maps and programs under its own filesystem
+`/sys/fs/bpf`. Users are able to make each object visible under the bpffs.
+We call it `object pinning`. This is done by calling syscall `bpf(2)` with
+a command `BPF_OBJ_PIN`. After doing that, users are able to use the object
+with commands such as `BPF_OBJ_GET`, or remove the object with an ordinary
+VFS syscall `unlink(2)`.
+
+Doing that, we can make maps and programs stay alive across process
+terminations. This mechanism provides a much more consistent way of sharing
+object with other processes, compared to other solutions such as `tc`,
+where objects are shared via Unix domain sockets.
+
+## Different pinning options
+
+`C.bpf_map_def.pinning` can be set to one the following pinning options.
+
+* `PIN_NONE` : object is not pinned
+* `PIN_OBJECT_NS` : pinning that is local to an object (to-be-implemented)
+* `PIN_GLOBAL_NS` : pinning with a global namespace under e.g. `/sys/fs/bpf/ns1/globals`
+* `PIN_CUSTOM_NS` : pinning with a custom path given as section parameter
+
+### pinning with `PIN_CUSTOM_NS`
+
+When loading a module with `C.bpf_map_def.pinning` set to `PIN_CUSTOM_NS`,
+an additional path must be set to `elf.SectionParams.PinPath` as parameter
+to `Load()`. For example:
+
+```
+/*
+#include <linux/bpf.h>
+
+struct bpf_map_def SEC("maps/dummy_array_custom") dummy_array_custom = {
+	.type = BPF_MAP_TYPE_ARRAY,
+	.key_size = sizeof(int),
+	.value_size = sizeof(unsigned int),
+	.max_entries = 1024,
+	.pinning = PIN_CUSTOM_NS,
+	.namespace = "ns1"
```

Comment:
is the `namespace` field used at all when the `PinPath` is specified below?

Reply:
@alban I'm not sure about that, but without namespace, the map cannot be loaded at all. That seems to be like that when the map definition has its own pinning option.

Reply:
For me, tests pass without; maybe the error was due to something else?

Reply:
@schu True. Now I cannot reproduce that error. @.@ I'll remove the namespace from the document.